com.unity.robotics.urdf-importer/Runtime/MeshProcessing/StlAssetPostProcessor.cs

```diff
@@ -13,18 +13,30 @@
 */ 
 
 using UnityEngine;
-using UnityEditor;
 using System.Linq;
 using System.IO;
 
 namespace RosSharp
 {
+    using Unity.Robotics;
+#if UNITY_EDITOR
+    using UnityEditor;
     public class StlAssetPostProcessor : AssetPostprocessor
+#else
+    public class StlAssetPostProcessor
+#endif    
     {
         private static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromPath)
         {
-            foreach (string stlFile in importedAssets.Where(x => x.ToLowerInvariant().EndsWith(".stl")))
-                createStlPrefab(stlFile);
+#if UNITY_EDITOR
+            if (!RuntimeURDF.IsRuntimeMode())
```

Comment:
Seen this in many places. Why do we need a check for runtime mode if we are in editor ?

Reply:
a few reasons:

- To be able to run and debug the run-time loading code also in editor.
- The run-time import code can also be used in editor to load URDF files that are not in the Assets, so if we decide this is a useful feature, we just have to add a check in the import GUI to use the runtime mode when the selected URDF file is not in the Assets.
- Possible future plan to make all loading use the run-time mode, to reduce the code maintenance overhead.

Reply:
since checking the runtime mode happens in many places, would it be better to move this check to an earlier stage or a higher level so that it only needs to be checked once?

Reply:
I am not sure if we can do it for this specific function: https://docs.unity3d.com/ScriptReference/AssetPostprocessor.OnPostprocessAllAssets.html